### PR TITLE
Finalize implementation

### PR DESCRIPTION
**Pull request by felixZdi**
Adding remaining test cases and bug fixes for wildcard certificates and DNS delegation.

### CODE REVIEW

File: tests/integration/unhappy_path.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    client = AcmeDnsAzureClient(acme_config_manager.config)
+    results: [RotationResult] = client.issue_certificates()
+
+    ## Validate
+    for result in results:
+        assert result.result == CertbotResult.CREATED
+    cn, san = azure_key_vault_manager.get_cn_and_san_from_certificate(
+        key_vault_cert_name
+    )
+    # TODO double check: what is the expected CN here? Is this correct: CN=testautohappypath1.{domain}
```
Comment (felixZdi) [inline tests/integration/unhappy_path.py:278]: Please check

File: tests/integration/unhappy_path.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    )
+
+    ## Test
+    client = AcmeDnsAzureClient(acme_config_manager.config)
+    results: [RotationResult] = client.issue_certificates()
+
+    ## Validate
+    assert results[0].result == CertbotResult.CREATED
+
+    # TODO doublecheck if this behaviour is correct. Certbot throwing following error for shared TXT of 2 cname records:
```
Comment (felixZdi) [inline tests/integration/unhappy_path.py:343]: @rembik @CoolDuke could you please review this test case?

**Review by rembik — COMMENTED** [review]
Fine for me

File: tests/integration/unhappy_path.py (hunk 2 above)
Comment (rembik) [reply tests/integration/unhappy_path.py]: Reviewed and changed

File: tests/integration/unhappy_path.py (hunk 1 above)
Comment (rembik) [reply tests/integration/unhappy_path.py]: CN would be the the wirst domain. Second domain should be part of SAN

**Comment by rembik** [comment]
Looks good for now

**Review by rembik — APPROVED** [review]

**Pull request merged**